src/three-components/TextureUtils.ts

```diff
@@ -168,118 +171,140 @@ export default class TextureUtils extends EventDispatcher {
+   */
+  private[$loadSkyboxFromUrl](url: string, progressTracker?: ProgressTracker):
+      Promise<WebGLRenderTargetCube> {
+    if (!this[$skyboxCache].has(url)) {
+      const progressCallback =
+          progressTracker ? progressTracker.beginActivity() : () => {};
+      this[$skyboxCache].set(
+          url, this.loadEquirectAsCubeMap(url, progressCallback));
+    }
+
+    return this[$skyboxCache].get(url)!;
+  }
+
+  /**
+   * Loads a WebGLRenderTarget from a given URL. The render target in this
+   * case will be assumed to be used as an environment map.
+   */
+  private[$loadEnvironmentMapFromUrl](
+      url: string,
+      progressTracker?: ProgressTracker): Promise<WebGLRenderTarget> {
+    if (!this[$environmentMapCache].has(url)) {
+      const progressCallback =
+          progressTracker ? progressTracker.beginActivity() : () => {};
+      const environmentMapLoads =
+          this.loadEquirectAsCubeMap(url, progressCallback)
+              .then(interstitialEnvironmentMap => {
+                const environmentMap =
+                    this.pmremPass(interstitialEnvironmentMap.texture);
+                // In this case, we don't care about the interstitial
+                // environment map because it will never be used for anything,
+                // so dispose of it right away:
+                interstitialEnvironmentMap.dispose();
+                return environmentMap;
+              });
+
+      this[$environmentMapCache].set(url, environmentMapLoads);
+    }
+
+    return this[$environmentMapCache].get(url)!;
+  }
+
+  /**
+   * Loads a skybox from a given URL, then PMREM is applied to the
+   * skybox texture and the resulting WebGLRenderTarget is returned,
+   * with the assumption that it will be used as an environment map.
+   */
+  private[$loadEnvironmentMapFromSkyboxUrl](
+      url: string,
+      progressTracker?: ProgressTracker): Promise<WebGLRenderTarget> {
+    if (!this[$environmentMapCache].has(url)) {
+      const skyboxLoads = this[$loadSkyboxFromUrl](url, progressTracker);
+      const environmentMapLoads =
+          skyboxLoads.then(skybox => this.pmremPass(skybox.texture));
+
+      this[$environmentMapCache].set(url, environmentMapLoads);
+    }
+
+    return this[$environmentMapCache].get(url)!;
+  }
+
+  /**
+   * Loads a dynamically generated environment map.
+   */
+  private[$loadGeneratedEnvironmentMap](): Promise<WebGLRenderTarget> {
+    if (this[$generatedEnvironmentMap] == null) {
+      const environmentMapGenerator =
+          new EnvironmentMapGenerator(this.renderer);
+      const interstitialEnvironmentMap = environmentMapGenerator.generate();
+
+      this[$generatedEnvironmentMap] =
+          this.pmremPass(interstitialEnvironmentMap.texture);
+
+      // We should only ever generate this map once, and we will not be using
+      // the environment map as a skybox, so go ahead and dispose of all
+      // interstitial artifacts:
```

Comment:
Is it true that we don't use the generated environment map as a skybox? I'm not sure why that hadn't occurred to me before. Maybe I should update our fidelity tests accordingly.

Reply:
I think that it's fine to let the fidelity tests have a skybox. This is the existing behavior when specifying a `background-image`, so it is expected. We can change this behavior if we want 🤷‍♂ but let's file an issue to discuss it?